psi4/driver/driver.py

```diff
@@ -1196,22 +1187,49 @@ def optimize(name, **kwargs):
     molecule = kwargs.pop('molecule', core.get_active_molecule())
 
     # If we are freezing cartesian, do not orient or COM
-    if core.get_local_option("OPTKING", "FROZEN_CARTESIAN"):
+    if any([core.get_option('OPTKING', 'FROZEN_CARTESIAN'), core.get_option('OPTKING', 'EXT_FORCE_CARTESIAN')]):
+        if molecule.has_zmatrix():
+            raise ValidationError("Job includes cartesian coordinate constraints. This cannot be fully "
+                                  "obeyed due to zmatrix in input. Please convert your zmatrix to cartesian "
+                                  "coordinates if cartesian constraints are needed ")
```

Comment:
glad this is trapped

Reply:
I presume the issue here is that the user  does not know what the Cartesian geometry will be, and thus the user cannot have a clear conception of what the Cartesian constraints mean.  As far as optking is concerned, I used to just ignore the Z-matrix.  @AlexHeide , is this still what we are doing?  I agree the trap is sound regardless.

Reply:
Correct, the zmatrix still does not affect optking. This check was added because psi4 does not actually obey fix_com and fix_orientation when using zmatrices. By this I mean, psi4 doesn't keep the same orientation over the course of several geometry updates. I think psi4 converts our cartesians back to the zmatrix and then optking requests new cartesian coordinates forcing a zmatrix to Cartesian conversion which ignores the fix keywords.

The only thing the user is missing out on here is freezing atoms in cartesian space when using internal coordinates. Any other Cartesian constraints (like ranges), I agree, the user couldn't really hope to specify.